### PR TITLE
Support auto-detection of CubeResource types

### PR DESCRIPTION
**Pull request by ivanpopelyshev**
Alternative for #6405.

@GoodBoyDigital please confirm that it works for your project.

### CODE REVIEW

**Comment by bigtimebuddy** [comment]
Could you add a unit test for this? Just make sure we get a CubeResource back?

**Comment by bigtimebuddy** [comment]
we could use ArrayFixed in utils for an array of a defined length for CubeResourse content

File: packages/core/test/CubeResource.js [hunk 1]
```diff
@@ -11,4 +17,11 @@ describe('PIXI.resources.CubeResource', function ()
             new CubeResource(8);
         }).to.throw(Error, /invalid length/i);
     });
+    it('should be created through BaseTexture.from()', function ()
+    {
+        const path1 = this.baseTexUrl;
+        const baseTex = BaseTexture.from([path1, path1, path1, path1, path1, path1]);
+
+        expect(baseTex.resource instanceof CubeResource).to.be.true;
```
Comment (cursedcoder) [inline packages/core/test/CubeResource.js]: what about `expect(baseTex.resource).to.be.instanceof(CubeResource)`?

Comment (bigtimebuddy) [reply packages/core/test/CubeResource.js]: Yes, better!

**Comment by ivanpopelyshev** [comment]
Yep, I approve the fixed size in type.

**Pull request merged**